aten/src/ATen/native/cpu/QuickPartition.cpp

```diff
@@ -0,0 +1,572 @@
+// AVX2 optimized partition algorithm
+// Based on the following paper:
+// Shay Gueron, Vlad Krasnov, Fast Quicksort Implementation Using AVX Instructions, The Computer Journal, Volume 59, Issue 1, January 2016, Pages 83–90, https://doi.org/10.1093/comjnl/bxv063
```

Comment:
Why not link to an actual PDF? E.g. http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.1009.7773&rep=rep1&type=pdf